### ISSUE TITLE
home-assistant doesn't build on master/unstable channels

### ISSUE THREAD

**Issue by etu** [NixOS/nixpkgs#66510]
**Describe the bug**
Home-assistant doesn't build on master or unstable channels.

**To Reproduce**
1. clone master
2. nix-build -A home-assistant
3. see it fail

**Expected behavior**
Build the package.

**Additional context**
I did a bit of git bisecting on this and came the conclusion that 3f19225e6f872585c051df9de85be1ae2248382f broke home-assistant.

The build failure is on the last test, which happens to be the yaml test:
```
tests/util/test_unit_system.py ................                          [ 96%]
tests/util/test_volume.py .....                                          [ 96%]
tests/util/test_yaml.py ..............................                   [100%]

=================================== FAILURES ===================================
_____________________ test_async_handler_loop_log[pyloop] ______________________

loop = <_UnixSelectorEventLoop running=False closed=False debug=False>

    @asyncio.coroutine
    def test_async_handler_loop_log(loop):
        """Test the logging sensitive data filter."""
        loop._thread_ident = threading.get_ident()

        queue = asyncio.Queue(loop=loop)
        base_handler = logging.handlers.QueueHandler(queue)
        handler = logging_util.AsyncHandler(loop, base_handler)

        # Test passthrough props and noop functions
        assert handler.createLock() is None
        assert handler.acquire() is None
        assert handler.release() is None
        assert handler.formatter is base_handler.formatter
        assert handler.name is base_handler.get_name()
        handler.name = 'mock_name'
        assert base_handler.get_name() == 'mock_name'

        log_record = logging.makeLogRecord({'msg': "Test Log Record"})
        handler.emit(log_record)
        yield from handler.async_close(True)
>       assert queue.get_nowait() == log_record
E       assert <LogRecord: None, None, , 0, "Test Log Record"> == <LogRecord: None, None, , 0, "Test Log Record">
E        +  where <LogRecord: None, None, , 0, "Test Log Record"> = <bound method Queue.get_nowait of <Queue at 0x7ffff3005790 maxsize=0 tasks=1>>()
E        +    where <bound method Queue.get_nowait of <Queue at 0x7ffff3005790 maxsize=0 tasks=1>> = <Queue at 0x7ffff3005790 maxsize=0 tasks=1>.get_nowait

tests/util/test_logging.py:43: AssertionError
---------------------------- Captured stderr setup -----------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------ Captured log setup ------------------------------
DEBUG    asyncio:selector_events.py:53 Using selector: EpollSelector
____________________ test_async_handler_thread_log[pyloop] _____________________

loop = <_UnixSelectorEventLoop running=False closed=False debug=False>

    @asyncio.coroutine
    def test_async_handler_thread_log(loop):
        """Test the logging sensitive data filter."""
        loop._thread_ident = threading.get_ident()

        queue = asyncio.Queue(loop=loop)
        base_handler = logging.handlers.QueueHandler(queue)
        handler = logging_util.AsyncHandler(loop, base_handler)

        log_record = logging.makeLogRecord({'msg': "Test Log Record"})

        def add_log():
            """Emit a mock log."""
            handler.emit(log_record)
            handler.close()

        yield from loop.run_in_executor(None, add_log)
        yield from handler.async_close(True)

>       assert queue.get_nowait() == log_record
E       assert <LogRecord: None, None, , 0, "Test Log Record"> == <LogRecord: None, None, , 0, "Test Log Record">
E        +  where <LogRecord: None, None, , 0, "Test Log Record"> = <bound method Queue.get_nowait of <Queue at 0x7ffff3071590 maxsize=0 tasks=1>>()
E        +    where <bound method Queue.get_nowait of <Queue at 0x7ffff3071590 maxsize=0 tasks=1>> = <Queue at 0x7ffff3071590 maxsize=0 tasks=1>.get_nowait

tests/util/test_logging.py:66: AssertionError
---------------------------- Captured stderr setup -----------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------ Captured log setup ------------------------------
DEBUG    asyncio:selector_events.py:53 Using selector: EpollSelector
============= 2 failed, 941 passed, 1 deselected in 37.54 seconds ==============
builder for '/nix/store/hy57m11jgza37n70h60lw56jlbc5l8yj-homeassistant-0.93.2.drv' failed with exit code 1
error: build of '/nix/store/hy57m11jgza37n70h60lw56jlbc5l8yj-homeassistant-0.93.2.drv' failed
```

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.63, NixOS, 19.09pre187507.525eaf407d4 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"home-manager, nixos-19.09pre187507.525eaf407d4"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: home-assistant
# a list of nixos modules affected by the problem
module:
```

**Comment by etu** [comment]
cc @FRidh @dotlambda

**Comment by etu** [comment]
Oh, so I just found this comment: https://github.com/NixOS/nixpkgs/pull/64293#issuecomment-511376435

So it will probably be fixed together with other things that are broken since that update.

**Comment by bjornfor** [comment]
Same error on the stable branch (release-19.03). When the fix lands it'd be a good idea to backport.

**Comment by lheckemann** [comment]
While the error is the same, it seems that the fix will probably look different, since it was introduced by a python update there… see the issue above.

**Comment by FRidh** [comment]
libyaml 0.2.1 to 0.2.2 bump changed certain behavior affecting pyyaml. I suppose this in the end results in the issue we see here. This is for upstream to resolve.

**Comment by lheckemann** [comment]
This is fixed by #65128, which is now in staging.
